src/cmd/services/m3aggregator/config/aggregator.go

```diff
@@ -846,3 +868,91 @@ func setMetricPrefix(
 	}
 	return fn([]byte(*str))
 }
+
+// passThroughWriter writes passthrough metrics to backends.
+type passThroughWriter struct {
+	numShards int
+	writers   []writer.Writer
```

Comment:
can you make a new writer.Writer w/ the locking inside it? will make this simpler to read.